runtime/src/main/java/org/corfudb/runtime/DistributedCheckpointerHelper.java

```diff
@@ -31,40 +31,52 @@ public static enum UpdateAction {
     public void updateCompactionControlsTable(Table<StringKey, RpcCommon.TokenMsg, Message> checkpointTable,
                                               StringKey stringKey,
                                               UpdateAction action) {
+
         try (TxnContext txn = corfuStore.txn(CORFU_SYSTEM_NAMESPACE)) {
-            if (action == UpdateAction.PUT) {
-                txn.putRecord(checkpointTable, stringKey,
-                        RpcCommon.TokenMsg.newBuilder().setSequence(System.currentTimeMillis()).build(), null);
-            } else if (action == UpdateAction.DELETE) {
-                txn.delete(checkpointTable, stringKey);
-            }
+            updateCompactionControlsTable(txn, checkpointTable, stringKey, action);
             txn.commit();
         } catch (Exception e) {
```

Comment:
I think we should throw the exception so that the caller can retry the update operation.